core/src/lib.rs

```diff
@@ -145,89 +186,69 @@ mod tests {
-            .or_insert(Value::String("Permanent Resident Card".to_string()));
-        kv_body
-            .entry("description".to_string())
-            .or_insert(Value::String(
-                "Government of Example Permanent Resident Card.".to_string(),
-            ));
-        kv_body
-            .entry("issuanceDate".to_string())
-            .or_insert(Value::String("2019-12-03T12:19:52Z".to_string()));
-        kv_body
-            .entry("expirationDate".to_string())
-            .or_insert(Value::String("2029-12-03T12:19:52Z".to_string()));
-
-        kv_subject
-            .entry("id".to_string())
-            .or_insert(Value::String("did:example:b34ca6cd37bbf23".to_string()));
-
-        let type_rs = serde_json::to_value(["PermanentResident".to_string(), "Person".to_string()]);
-        if type_rs.is_ok() {
-            kv_subject
-                .entry("type".to_string())
-                .or_insert(type_rs.unwrap());
-        }
-
-        kv_subject
-            .entry("givenName".to_string())
-            .or_insert(Value::String("JOHN".to_string()));
-        kv_subject
-            .entry("familyName".to_string())
-            .or_insert(Value::String("SMITH".to_string()));
-        kv_subject
-            .entry("gender".to_string())
-            .or_insert(Value::String("Male".to_string()));
-        kv_subject
-            .entry("image".to_string())
-            .or_insert(Value::String(
-                "data:image/png;base64,iVBORw0KGgo...kJggg==".to_string(),
-            ));
-        kv_subject
-            .entry("residentSince".to_string())
-            .or_insert(Value::String("2015-01-01".to_string()));
-        kv_subject
-            .entry("lprCategory".to_string())
-            .or_insert(Value::String("C09".to_string()));
-        kv_subject
-            .entry("lprNumber".to_string())
-            .or_insert(Value::String("999-999-999".to_string()));
-        kv_subject
-            .entry("commuterClassification".to_string())
-            .or_insert(Value::String("C1".to_string()));
-        kv_subject
-            .entry("birthCountry".to_string())
-            .or_insert(Value::String("Bahamas".to_string()));
-        kv_subject
-            .entry("birthDate".to_string())
-            .or_insert(Value::String("1958-07-17".to_string()));
+        let (kv_body, kv_subject) = get_body_subject();
 
         let vc = to.create_credential(
             crate::CRED_TYPE_PERMANENT_RESIDENT_CARD.to_string(),
             kv_subject,
             kv_body,
             "https://issuer.oidp.uscis.gov/credentials/83627465",
         );
+
         assert!(vc.is_ok());
-        assert_json_eq!(expect, vc.unwrap());
+        let credential = vc.unwrap();
+        assert_json_eq!(expect_credential, credential.serialize());
+
+        test_create_presentation(to, credential)
+    }
+
+    fn test_create_presentation(to: TestObj, credential: crate::Credential) -> Result<(), String> {
```

Comment:
unless this is a unit test we probably shouldn't add a `test_` prefix to the function.